src/main.ts

```diff
@@ -0,0 +1,51 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { Octokit } from "@octokit/rest";
+import { Value } from "@sinclair/typebox/build/cjs/value";
+import { envSchema, pluginSettingsSchema, PluginInputs } from "./types";
+import { plugin } from "./plugin";
+
+/**
+ * How a GitHub action executes the plugin.
+ */
+export async function run() {
+  const payload = github.context.payload.inputs;
+
+  const env = Value.Decode(envSchema, process.env);
+  const settings = Value.Decode(pluginSettingsSchema, JSON.parse(payload.settings));
+
+  const inputs: PluginInputs = {
+    stateId: payload.stateId,
+    eventName: payload.eventName,
+    eventPayload: JSON.parse(payload.eventPayload),
+    settings,
+    authToken: env.UBIQUIBOT_TOKEN,
```

Comment:
Should come from the payload

Reply:
my mistake didn't realize I had overlooked this